### ISSUE TITLE
`new-pane --cwd` doesn't use the given cwd

### ISSUE THREAD

**Issue by p00f** [zellij-org/zellij#2766]
Thank you for taking the time to file this issue! Please follow the instructions and fill in the missing parts below the instructions, if it is meaningful. Try to be brief and concise.

**Basic information**

`zellij --version`: `zellij 0.38.1`
`stty size`: 36 174
`uname -av` or `ver`(Windows): `Linux ganymede 6.4.12-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 24 Aug 2023 00:38:14 +0000 x86_64 GNU/Linux`

List of programs you interact with as, `PROGRAM --version`: output cropped meaningful, for example: N/A

**Further information**
Steps to reproduce:
1) `zellij`
2) `zellij action new-pane --cwd /home/p00f/foo`

Expected behaviour:
New pane with `~/foo` as the cwd

Actual behaviour:
New pane with `~` as the cwd (doesn't matter what the cwd of the old pane was, it starts from `~`)

**Comment by marianomarciello** [comment]
Hi I think this is related on how the `cwd` argument is used in [this](https://github.com/zellij-org/zellij/blob/697723ddd30715e2997ad12352ea3c89ebdb7e17/zellij-utils/src/input/actions.rs#L302) function.
The `cwd` parameter is joined with the `current_dir`. If this join operation is not successful then`current_dir` is returned.

I think this needs to be extended as follow:
1. check if `cwd` is an absolute path or a relative path
2. if `cwd` is an absolute path then use it as path for the new Panel
3. if `cwd` is a  relative path then join it with `current_dir`
4. otherwise use `current_dir` (`cwd`  is empty)

Also, this need to be modified for other functions too (like: NewTab [link](https://github.com/zellij-org/zellij/blob/697723ddd30715e2997ad12352ea3c89ebdb7e17/zellij-utils/src/input/actions.rs#L406) )

**Comment by p00f** [comment]
Yes - https://github.com/zellij-org/zellij/blob/697723ddd30715e2997ad12352ea3c89ebdb7e17/zellij-utils/src/input/actions.rs#L356C

The Option<RunCommandAction> (second element in the tuple) should be a RunCommandAction with the given cwd: https://github.com/zellij-org/zellij/blob/697723ddd30715e2997ad12352ea3c89ebdb7e17/zellij-utils/assets/prost/api.action.rs#L239

**Comment by marianomarciello** [comment]
I think we need a way to cast a `TerminalAction` into a `RunCommandAction` as follow e.g. [link](https://github.com/zellij-org/zellij/blob/697723ddd30715e2997ad12352ea3c89ebdb7e17/zellij-server/src/plugins/zellij_exports.rs#L381C1-L384C7). In the previous example the cast is done on the server-side.

I was unable to find the same conversion on the client-side.
On the server side the conversion can be done using the function `get_default_shell` as follow [link](https://github.com/zellij-org/zellij/blob/697723ddd30715e2997ad12352ea3c89ebdb7e17/zellij-server/src/pty.rs#L474C1-L481C3),
Should we insert a `get_default_shell` function on the client-side too ?
Are there some other ways to get around this problem ?

**Comment by KasimKaizer** [comment]
I face a similar issue where the --cwd doesn't work.

Command:
`zellij action new-tab --name "New Tab" --cwd "$PWD" -l default`

This should (according to the docs) create a new tab window with the current directory, but instead it uses the directory from where zellij session was originally started.

**Comment by bennyyip** [comment]
This issue still exists in 0.41.2
`cwd` is not use in the `else` branch.
https://github.com/zellij-org/zellij/blob/v0.41.2/zellij-utils/src/input/actions.rs#L454-L466

**Comment by DarioDarko** [comment]
still the same in 0.43.1
the workaround above works tho

**Comment by Sawangg** [comment]
Until this gets fixed, another workaround to create a new tab with your cwd (working in 0.43.1) 

```bash
 zellij action new-tab
 zellij action write-chars "cd <your cwd> && clear"
 zellij action write 10
```
